File: domain/src/main/java/org/openmedstack/eventstore/SagaRepository.java
package org.openmedstack.eventstore;

import org.openmedstack.domain.Saga;

import java.util.HashMap;
import java.util.concurrent.CompletableFuture;
import java.util.function.Consumer;

public interface SagaRepository {
    <TSaga extends Saga> CompletableFuture<TSaga> getById(Class<TSaga> type, String sagaId);

    CompletableFuture<Boolean> save(Saga saga, Consumer<HashMap<String, Object>> updateHeaders);
}

File: domain/src/main/kotlin/org/openmedstack/eventstore/SagaRepository.kt
package org.openmedstack.eventstore

import org.openmedstack.domain.Saga
import java.util.concurrent.CompletableFuture
import java.util.function.Consumer

interface SagaRepository {
    fun <TSaga : Saga> getById(type: Class<TSaga>, sagaId: String): CompletableFuture<TSaga>
    fun save(saga: Saga, updateHeaders: Consumer<HashMap<String, Any>>): CompletableFuture<Boolean>
}
